python/pyspark/ml/linalg/__init__.py

```diff
@@ -976,14 +976,20 @@ def __getitem__(self, indices):
             return self.values[i + j * self.numRows]
 
     def __eq__(self, other):
+        def _eq(self, other):
```

Comment:
Now `_eq` won't be called in multiple places, we can remove the `_eq` definition, make code simpler.